packages/cli/generation/local-generation/local-workspace-runner/src/runLocalGenerationForWorkspace.ts

```diff
@@ -119,7 +124,25 @@ async function writeFilesToDiskAndRunGenerator({
     if (firstLocalOutputItem == null) {
         return;
     }
-    await rm(absolutePathToLocalOutput, { force: true, recursive: true });
+    // await rm(absolutePathToLocalOutput, { force: true, recursive: true });
+    const absolutePathToFernignore = path.join(absolutePathToLocalOutput, ".fernignore");
+    let filesToBeIgnoredByFern: string[] = [".fernignore"];
+    const absolutePathToFernignoreExists = await doesPathExist(AbsoluteFilePath.of(absolutePathToFernignore));
+
+    if (absolutePathToFernignoreExists) {
+        const absolutePathToFernignoreContent = await readFile(absolutePathToFernignore, "utf-8");
+        filesToBeIgnoredByFern = filesToBeIgnoredByFern.concat(absolutePathToFernignoreContent.trim().split(/\r?\n/));
+        filesToBeIgnoredByFern = filesToBeIgnoredByFern.filter(filePattern => filePattern !== "" && !filePattern.startsWith("#"));
+        filesToBeIgnoredByFern = globSync(filesToBeIgnoredByFern, {
+            cwd: absolutePathToLocalOutput, root: "", dot: true, nobrace: false, noext: false, matchBase: true
+        });
+        await EXEC("git init", {cwd: absolutePathToLocalOutput});
+        await EXEC("git add .", {cwd: absolutePathToLocalOutput});
+        await EXEC("git commit -m \"initial\"", {cwd: absolutePathToLocalOutput});
+        await EXEC("git rm -rf .", {cwd: absolutePathToLocalOutput});
```

Comment:
what happens if any of these commands fails? i think we should catch the exception and throw

Reply:
Great. So, what in the exception should be thrown? Just the message?